grails-app/controllers/com/unifina/controller/api/IntegrationKeyApiController.groovy

```diff
@@ -65,6 +73,25 @@ class IntegrationKeyApiController {
 		render ""
 	}
 
+	@StreamrApi
+	def update(String id) {
+		Map json = new JsonSlurper().parseText((String) request.JSON)
+		if (json.name == null || json.name.trim() == "") {
+			response.status = 400
```

Comment:
It would be good to return a JSON error here in the format defined in `com.unifina.api.ApiError`. One way to do it is throw an `com.unifina.api.ApiException` here after which `com.unifina.controller.api.ErrorController` will handle setting the status code and responding with JSON automagically.

So for example `throw new ApiException(400, "BODY_NOT_VALID", "must send JSON body containing property 'name'.")`